### PR TITLE
GetLegalEntity

### PR DESCRIPTION
**Pull request by PeterDavidWilliams**

### CODE REVIEW

File: src/SFA.DAS.EAS.Web.IntegrationTests/LegalEntitiesControllerTests/WhenIGetLegalEntitiesWithKnownIds.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            var callRequirements = new CallRequirements($"api/accounts/{hashedAccountId}/legalentities")
+                .ExpectControllerType(typeof(LegalEntitiesController))
+                .AllowStatusCodes(HttpStatusCode.OK);
+
+            // Act
+            var account = await _tester.InvokeGetAsync<ResourceList>(callRequirements);
+
+            // Assert
+            Assert.IsNotNull(account.Data);
+ //           account.Data[0];
```
Comment (robinoconnor) [inline src/SFA.DAS.EAS.Web.IntegrationTests/LegalEntitiesControllerTests/WhenIGetLegalEntitiesWithKnownIds.cs]: this commented out part needs to be removed

File: src/SFA.DAS.EAS.Web.IntegrationTests/LegalEntitiesControllerTests/WhenIGetLegalEntitiesWithUnknownIds.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        public async Task ThenTheStatusShouldBeNotFound_ByHashedId()
+        {
+            // Arrange
+            var callRequirements = new CallRequirements($"api/accounts/MADE*UP*ID/legalentities");
+
+            // Act
+            await _tester.InvokeGetAsync<ResourceList>(callRequirements);
+
+            // Assert
+            Assert.Pass("Verified we got http status NotFound");
```
Comment (DScuffle) [inline src/SFA.DAS.EAS.Web.IntegrationTests/LegalEntitiesControllerTests/WhenIGetLegalEntitiesWithUnknownIds.cs]: We don't actually check for not found so I don't think we can assert a Pass here - previously we had a verification set up when creating the callRequirements object. In other tests we are doing something like account.ExpectStatusCodes(HttpStatusCode.OK);

Comment (iangledhill) [reply src/SFA.DAS.EAS.Web.IntegrationTests/LegalEntitiesControllerTests/WhenIGetLegalEntitiesWithUnknownIds.cs]: They're extension methods on the return type from InvokeGetAsync<>() so if you assign the result to a local variable then you should see the available methods in intellisense.

File: src/SFA.DAS.EAS.Web.IntegrationTests/LegalEntitiesControllerTests/WhenIGetLegalEntitiesWithKnownIds.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        [Test]
+        public async Task ThenTheStatusShouldBeFound_ByHashedAccountId()
+        {
+            var callRequirements = new CallRequirements($"api/accounts/{_employerAccount.HashedAccountId}/legalentities");
+
+            // Act
+            var account = await _tester.InvokeGetAsync<ResourceList>(callRequirements);
+
+            // Assert
+            Assert.IsNotNull(account.Data);
```
[the remark this answers is not in the document]
Comment (PeterDavidWilliams) [reply src/SFA.DAS.EAS.Web.IntegrationTests/LegalEntitiesControllerTests/WhenIGetLegalEntitiesWithKnownIds.cs]: Changes made - added count check and separated single legal entity and multiple legal entity tests out.

File: src/SFA.DAS.EAS.Web.IntegrationTests/LegalEntitiesControllerTests/WhenIGetLegalEntitiesWithUnknownIds.cs (hunk 2 above)
Comment (PeterDavidWilliams) [reply src/SFA.DAS.EAS.Web.IntegrationTests/LegalEntitiesControllerTests/WhenIGetLegalEntitiesWithUnknownIds.cs]: Changes made as requested.

**Pull request merged**